cmd/argocd/commands/app.go

```diff
@@ -473,6 +475,33 @@ func setHelmOpt(src *argoappv1.ApplicationSource, valueFiles []string, releaseNa
 	}
 }
 
+func setJsonnetOpt(src *argoappv1.ApplicationSource, tlaParameters []string) {
+	if src.Directory == nil {
+		src.Directory = &argoappv1.ApplicationSourceDirectory{}
+	}
+
+	if len(tlaParameters) != 0 {
+		tlas := make([]argoappv1.JsonnetVar, len(tlaParameters))
+		for index, paramStr := range tlaParameters {
+			parts := strings.SplitN(paramStr, "=", 2)
+			if len(parts) != 2 {
```

Comment:
not sure you need this check?

Reply:
it has to have the format parameter=value, so if I don't get it like this, then I can't continue
and the same is done for helm, ksonnet params and setting labels

Reply:
ah - of course - it is at most 2 parts, not exactly 2 parts